src/rpc/blockchain.cpp

```diff
@@ -1452,7 +1452,7 @@ UniValue preciousblock(const JSONRPCRequest& request)
     return NullUniValue;
 }
 
-UniValue invalidateblock(const JSONRPCRequest& request)
+static UniValue invalidateblock(const JSONRPCRequest& request) EXCLUSIVE_LOCKS_REQUIRED(::mempool.cs)
```

Comment:
All the locks are taken. I don't see why this is necessary.

Reply:
@MarcoFalke `invalidateblock(…)` calls `InvalidateBlock(…)` without first locking `::mempool.cs`. Calling `InvalidateBlock(…)` requires holding mutex `mempool.cs` exclusively. Then it follows that calling `invalidateblock(…)` should require `::mempool.cs` too which is what the annotations says? What am I missing? :-)

Reply:
> Calling InvalidateBlock(…) requires holding mutex mempool.cs

Yeah, I looked and it seems all the locks are taken here, which one did I miss?

Reply:
Ah, sorry. The `InvalidateBlock(…)` annotation seems incorrect. Thanks!